### ISSUE TITLE
Live config reloading + logging level

### ISSUE THREAD

**Issue by prawnsalad** [DanielOaks/gircbnc#26]
Most indepth logging level can include full raw log output

**Comment by prawnsalad** [comment]
Probably linked with https://github.com/goshuirc/bnc/issues/15

**Comment by DanielOaks** [comment]
Yeah this is the sorta stuff I wanna get accomplished by copying Oragono's logging. The way Oragono does it is flexible and quite nice imo, so it'd be cool to clone that directly. I can migrate this over for the 'core' and then we can pass it along to the components as well.

It should be kept in mind, logging for history (ala [message logging](https://github.com/goshuirc/bnc/blob/master/lib/components/messageLogger/filelogger.go)) and logging to a log file / to the terminal are completely separate. We can continue with the current differentiator by just calling the former "message logging" and this... "debug logging" or something similar.
